packages/type-utils/typings/typescript.d.ts

```diff
@@ -36,6 +36,8 @@ declare module 'typescript' {
      * Maps from a SourceFile's `.path` to the name of the package it was imported with.
      */
     readonly sourceFileToPackageName: ReadonlyMap<Path, string>;
+
+    readonly resolvedModules: ReadonlyMap<Path, unknown>;
```

Comment:
And this..... the typescript compiler source code has a very complex type for the value of the Map.

Thoughts about how to improve this ?

Reply:
https://github.com/typescript-eslint/typescript-eslint/pull/8502#discussion_r1541735957